aero.minova.core.application.system.service/src/main/java/aero/minova/core/application/system/controller/SqlProcedureController.java

```diff
@@ -135,184 +137,140 @@ public SqlProcedureResult calculateSqlProcedureResult(Table inputTable, List<Row
+									} else if (type == Types.NVARCHAR) {
+										return new Column(name, DataType.STRING);
+									} else if (type == Types.DECIMAL) {
+										return new Column(name, DataType.BIGDECIMAL);
+									} else {
+										throw new UnsupportedOperationException("msg.UnsupportedResultSetError %" + i);
+									}
+								} catch (Exception e) {
+									throw new RuntimeException("msg.ParseResultSetError");
 								}
-							} catch (Exception e) {
-								throw new RuntimeException("msg.ParseResultSetError");
+							}).collect(toList()));
+					int totalResults = 0;
+
+					int securityTokenInColumn = findSecurityTokenColumn(resultSet);
+					resultSet.setMetaData(new TableMetaData());
+					while (sqlResultSet.next()) {
+						Row rowToBeAdded = null;
+						if (limit > 0) {
+							// nur die Menge an Rows, welche auf der gewünschten Page liegen
+							if (sqlResultSet.getRow() > ((page - 1) * limit) && sqlResultSet.getRow() <= (page * limit)) {
+								rowToBeAdded = convertSqlResultToRow(resultSet//
+										, sqlResultSet//
+										, customLogger.logger////
+										, this);
 							}
-						}).collect(toList()));
-				int totalResults = 0;
-
-				int securityTokenInColumn = findSecurityTokenColumn(resultSet);
-				resultSet.setMetaData(new TableMetaData());
-				while (sqlResultSet.next()) {
-					Row rowToBeAdded = null;
-					if (limit > 0) {
-						// nur die Menge an Rows, welche auf der gewünschten Page liegen
-						if (sqlResultSet.getRow() > ((page - 1) * limit) && sqlResultSet.getRow() <= (page * limit)) {
+						} else {
 							rowToBeAdded = convertSqlResultToRow(resultSet//
 									, sqlResultSet//
 									, customLogger.logger////
 									, this);
 						}
-					} else {
-						rowToBeAdded = convertSqlResultToRow(resultSet//
-								, sqlResultSet//
-								, customLogger.logger////
-								, this);
-					}
 
-					/*
-					 * Falls die SecurityToken-Prüfung nicht eingeschalten ist, wird einfach true zurückgegeben und die Row hinzugefügt.
-					 */
-					if (checkRowForValidSecurityToken(userSecurityTokensToBeChecked, rowToBeAdded, securityTokenInColumn)) {
-						resultSet.addRow(rowToBeAdded);
-						totalResults++;
+						/*
+						 * Falls die SecurityToken-Prüfung nicht eingeschalten ist, wird einfach true zurückgegeben und die Row hinzugefügt.
+						 */
+						if (checkRowForValidSecurityToken(userSecurityTokensToBeChecked, rowToBeAdded, securityTokenInColumn)) {
+							resultSet.addRow(rowToBeAdded);
+							totalResults++;
+						}
+						resultSet.fillMetaData(resultSet, limit, totalResults, page);
 					}
-					resultSet.fillMetaData(resultSet, limit, totalResults, page);
 				}
-			}
-			// Dies muss ausgelesen werden, nachdem die ResultSet ausgelesen wurde, da sonst diese nicht abrufbar ist.
-			val returnCode = preparedStatement.getObject(1);
-			if (returnCode != null) {
-				result.setReturnCode(preparedStatement.getInt(1));
```

Comment:
Hier bräuchten wir eine Liste von return codes.

Reply:
Welcher ReturnCode soll am Ende zurückgegeben werden? Man kann nur einen schicken.

Reply:
Das müsste man so abändern, dass man auch eine Liste zurückschicken kann.
Dazu könnte man den Wert returnCodes zurückgeben.
In returnCode würde man 0 reinschreiben,
wenn alle returnCodes 0 sind,
und ansonsten 1.